sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/betweenExpression.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.expressions
+
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = "BETWEEN(projection, lower, upper) - Returns true if `projection` > `lower` and `projection` <  `upper`.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(0.5, 0, 1);
+        TRUE
+  """,
+  since = "2.0.0",
+  group = "conditional_funcs")
+case class BetweenExpr(proj: Expression, lower: Expression,
+                       upper: Expression, replacement: Expression)
```

Comment:
Please, adjust indentations according to https://github.com/databricks/scala-style-guide?tab=readme-ov-file#indent